services/user/Chainmail/ui/src/components/mail-display.tsx

```diff
@@ -91,33 +92,138 @@ const ActionBar = ({
         deleteDraftById,
     } = useDraftMessages();
 
-    const onDelete = () => {
-        if (mailbox === "inbox") {
-            setInboxMessageId("");
-        } else if (mailbox === "drafts") {
-            setDraftMessageId("");
-            if (!selectedDraftMessage?.id) return;
-            deleteDraftById(selectedDraftMessage.id);
-        }
+    const onArchive = async (itemId: string) => {
+        let id = parseInt(itemId);
+        const supervisor = await getSupervisor();
+        // TODO: Unsave if saved
+        // await supervisor.functionCall({
+        //     service: "chainmail",
+        //     intf: "api",
+        //     method: "unsave",
+        //     params: [id],
+        // });
+        // TODO: Improve error detection. This promise resolves with success before the transaction is pushed.
```

Comment:
Is it? The tx is not _applied_ before promise resolution, but the tx is supposed to be pushed as part of the resolution of the plugin call (before resolving this promise).